### PR TITLE
Add configuration option to change date format

### PR DESCRIPTION
**Pull request by da-moo**
Hello! Just a simple pull request to add the option to configure date formats to whatever the user wants. I tried to do my due diligence and show the changes in both the `README.md` and example site's `config.toml`. Let me know if you need anything changed. Thank you!

### CODE REVIEW

**Comment by panr** [comment]
@da-moo Looks good, but... Do we really need two separated variables for list and post?  🤔

**Comment by da-moo** [comment]
Hey @panr you're probably right, we don't need two separate variables but during my testing I found that I liked having a shortened date in list format and a longer date in single format. I'm not opposed to changing the PR to just one variable, but I think two variables adds a tiny bit more flexibility. What do you think?

File: README.md [hunk 1]
```diff
@@ -97,6 +97,8 @@ paginate = 5
     newerPosts = "Newer posts"
     olderPosts = "Older posts"
     minuteReadingTime = "min read"
+    dateFormatSingle = "Monday, January 2, 2006"
```
Comment (panr) [inline README.md]: I would like to have "2006-01-02" by default, please 😉

File: README.md [hunk 2]
```diff
@@ -97,6 +97,8 @@ paginate = 5
...
+    dateFormatList = "Mon, Jan 2, 2006"
```
Comment (panr) [inline README.md]: As above

File: exampleSite/config.toml [hunk 3]
```diff
@@ -26,6 +26,8 @@ paginate = 5
     newerPosts = "Newer posts"
     olderPosts = "Older posts"
     minuteReadingTime = "min read"
+    dateFormatSingle = "Monday, January 2, 2006"
```
Comment (panr) [inline exampleSite/config.toml]: As above

File: exampleSite/config.toml [hunk 4]
```diff
@@ -26,6 +26,8 @@ paginate = 5
...
+    dateFormatList = "Mon, Jan 2, 2006"
```
Comment (panr) [inline exampleSite/config.toml]: As above

**Comment by da-moo** [comment]
Made the changes and squashed commits. Everything should be good to go

**Comment by panr** [comment]
@da-moo awesome! Thanks! 😎

**Pull request merged**